### PR TITLE
Barkin/cpstyle bugs

### PR DESCRIPTION
**Pull request by volfied**
Addresses #1977, #2006, and #2007 

Requires https://github.com/adobe-photoshop/spaces-adapter/pull/131

### CODE REVIEW

File: src/js/actions/shapes.js [hunk 1]
```diff
@@ -162,6 +162,58 @@ define(function (require, exports) {
             });
     };
 
+    var setStroke = function (document, layers, strokeIndex, stroke, enabled) {
```
Comment (iwehrman) [inline src/js/actions/shapes.js]: Missing JSdoc :neckbeard:

File: src/js/actions/shapes.js [hunk 2]
```diff
@@ -162,6 +162,58 @@ define(function (require, exports) {
...
+                    strokePromise,
+                    function () {
+                        return this.transfer(layerActions.resetBounds, document, layers);
+                    }.bind(this));
+        } else {
+            return layerActionsUtil.playSimpleLayerActions(document, layers, strokeObj, true, options)
+                .bind(this)
+                .then(function () {
+                    // upon completion, fetch the stroke info for all layers
+                    _refreshStrokes.call(this, document, layers, strokeIndex);
```
Comment (iwehrman) [inline src/js/actions/shapes.js]: The promise should be returned here.

**Comment by iwehrman** [comment]
2 nits 2 @volfied

**Comment by volfied** [comment]
@iwehrman addressed the nits, thank you

**Comment by iwehrman** [comment]
LGTM.

**Pull request merged**